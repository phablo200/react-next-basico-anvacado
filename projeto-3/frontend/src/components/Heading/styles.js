import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.xlarge};
    }
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${!colorDark ? theme.colors.white : theme.colors.primaryColor};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
