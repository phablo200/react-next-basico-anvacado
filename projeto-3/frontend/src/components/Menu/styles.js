import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${(props) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: red;
    border-bottom: ${props.theme.colors.mediumGray};
  `}
`;

export const Menu = styled.div`
  ${() => css``}
`;

export const Button = styled.div`
  ${() => css``}
`;
