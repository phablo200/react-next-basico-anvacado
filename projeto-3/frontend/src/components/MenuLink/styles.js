import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${(props) => css`
    display: block;
    text-decoration: none;
    font-size: ${props.theme.font.sizes.small};
    padding: ${props.theme.spacings.small};
    color: ${props.theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      left: 50%;
      bottom: 0.75rem;
      width: 0;
      height: 0.2rem;
      background: ${props.theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
