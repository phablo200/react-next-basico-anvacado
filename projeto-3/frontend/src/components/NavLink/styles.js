import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${(props) => css`
    display: flex;
    flex-flow: row wrap;
    @media ${props.theme.media.lteMedium} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
