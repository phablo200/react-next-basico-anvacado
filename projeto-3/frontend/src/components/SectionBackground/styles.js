import styled, { css } from 'styled-components';

const containerBackground = (theme) => css`
  ${console.log(theme)}
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${(props) => css`
    background: ${props.theme.colors.white};
    color: ${props.theme.colors.primaryColor};
    ${props.background && containerBackground(props.theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
