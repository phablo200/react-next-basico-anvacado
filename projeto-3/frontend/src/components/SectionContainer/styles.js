import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${(props) => css`
    max-width: 120rem;
    margin: 0 auto;
    padding: ${props.theme.spacings.large};
  `}
`;
