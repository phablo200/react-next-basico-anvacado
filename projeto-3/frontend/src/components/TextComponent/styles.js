import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${(props) => css`
    font-size: ${props.theme.font.sizes.medium};
  `}
`;
