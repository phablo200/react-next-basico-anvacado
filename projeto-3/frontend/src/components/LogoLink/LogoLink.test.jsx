import { renderTheme } from '../../styles/render-theme';
import { LogoLink } from '.';
import { screen } from '@testing-library/react';

describe('<LogoLink />', () => {
  it('Should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" />);
    const heading = screen.getByRole('heading', { name: 'Olá mundo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });

  it('Should render image logo', () => {
    renderTheme(<LogoLink link="#target" text="Olá mundo" srcImg="image.jpg" />);
    expect(screen.getByRole('img', { name: 'Olá mundo' })).toHaveAttribute('src', 'image.jpg');
    // expect screen.getByAltText('Olá mundo')
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<LogoLink link="#target" text="Olá mundo" srcImg="image.jpg" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
