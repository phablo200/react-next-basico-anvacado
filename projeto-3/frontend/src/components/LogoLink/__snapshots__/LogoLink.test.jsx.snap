// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> Should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 > img {
  max-height: 3.5rem;
}

.c0 {
  color: #0A1128;
  font-size: 5.6rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="OlÃ¡ mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
